### PR TITLE
Adds to signup component

### PR DESCRIPTION
**Pull request by Broveku**
PAIR PROGRAMMED WITH {{{{ Isaac Ellingwood }}}}

## Description

Adds to a basic functioning signup component {{WITHOUT ENDPOINT}}
Email, password, firsname, lastname
Base foundation for  form validation but is currently commented out

## Type of change

Please delete options that are not relevant.

- [x ] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have removed unnecessary comments/console logs from my code
- [ ] I have made corresponding changes to the documentation if necessary (optional)
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings
- [ ] No duplicate code left within changed files
- [ ] Size of pull request kept to a minimum
- [ ] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Comment by Broveku** [comment]
https://user-images.githubusercontent.com/84815843/139497866-9e5eea25-c2f6-46f7-865a-617c58b10519.mp4

VIDEO FOR SIGNUP COMPONENT

**Review by VAIBHAVIBALAR — APPROVED** [review]
Great work on signup form, with placeholder adding more light to it. Also the use of yup for form validation seems to be more precise and meeting good standards for the application.

**Review by jessejohns-1 — APPROVED** [review]
Everything looks great. yup looks good as well. I Will add to the list of questions to ensure yup is good with stakeholders. overall great work.

**Review by mariuscmoldovan — APPROVED** [review]
Looks good! great work.

File: src/components/pages/Signup/Signup.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      lastName: formValues.lastName.trim(),
+    };
+    console.log(newAccount);
+    postNewAccount(newAccount);
+  };
+
+  const onChange = e => {
+    console.log('radio checked', e.target.value);
+    setValue(e.target.value);
+  };
```
Comment (paulstgermain) [inline src/components/pages/Signup/Signup.js:63]: I'm assuming this is meant to be a placeholder until we connect with the back-end for asserting these roles?

File: src/components/pages/Signup/Signup.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    }
+  };
+
+  // const validate = (name, value) => {
+
+  //     reach(schema, name)
+  //         .validate(value)
+  //         .then(()=> setFormErrors({...formErrors, [name]: '' }))
+  //         .catch(err => setFormErrors({...formErrors, [name]: err.errors[0]}))
+  // }
```
Comment (paulstgermain) [inline src/components/pages/Signup/Signup.js:47]: I'm curious - I see all of our validation commented out, why is that?

File: src/components/pages/Signup/Signup.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  const [lastName, setLastName] = useState('');
+
+  const [formValues, setFormValues] = useState(initialFormValues);
+  const [formErrors, setFormErrors] = useState(initialFormErrors);
+
+  const [value, setValue] = React.useState(1);
+
+  const postNewAccount = async newAccount => {
+    try {
+      const response = await axios.post(' ', newAccount);
```
Comment (paulstgermain) [inline src/components/pages/Signup/Signup.js:34]: I'm assuming this is also a stub to represent where we'll be putting the actual call once we're ready to do so?

**Review by paulstgermain — APPROVED** [review]
## Video Walkthrough Review

Although short, this was a pretty detailed walkthrough, and I appreciate how you actually touched on the code itself! That said, I think it could be a little more detailed - having gone through the video and code alike I'm still left with a few questions:

1. What's the outstanding issue with form validation? (I do believe we should move forward with validation for our forms!)
2. Where is the form data going after submission? (An explanation of the stubbed-out Axios call would have been a great addition for context!)
3. Same as the last question for the role value from our form.

I get the context via experience, but it would be great practice to start developing an eye for these details when talking through your work - leave no questions in your interviewers' or managers' minds wherever possible!

## Code Review

As for the code, this is clean as it comes! It's well-written, well-formatted, and clearly functional. Given my understanding of the outstanding questions I left as examples, I'm comfortable approving this - great work overall!

**Pull request merged**